src/SDKs/EventGrid/DataPlane/Microsoft.Azure.EventGrid/Generated/Models/JobState.cs

```diff
@@ -11,130 +11,105 @@
 namespace Microsoft.Azure.EventGrid.Models
 {
     using Newtonsoft.Json;
+    using Newtonsoft.Json.Converters;
+    using System.Runtime;
+    using System.Runtime.Serialization;
 
     /// <summary>
     /// Defines values for JobState.
     /// </summary>
-    /// <summary>
-    /// Determine base value for a given allowed value if exists, else return
-    /// the value itself
-    /// </summary>
-    [JsonConverter(typeof(JobStateConverter))]
-    public struct JobState : System.IEquatable<JobState>
+    [JsonConverter(typeof(StringEnumConverter))]
+    public enum JobState
```

Comment:
Question to the reviewer: 

This model is based on the swagger definition in https://github.com/Azure/azure-rest-api-specs/blob/master/specification/eventgrid/data-plane/Microsoft.Media/stable/2018-01-01/MediaServices.json. There have been no changes in this swagger since May 18, hence I am looking to understand why this got re-generated (new autorest behavior?). Can you please help me understand this?

Reply:
@kalyanaj The behavior per se would be the same for both ends, i.e., the REST service sending/receiving this enum value vs the client.
Its a better/cleaner implementation of extensible enum that we released over the past few months. We do have some flags available to revert to the legacy implementation.

Reply:
Thanks @dsgouda  for the info!